internal/services/batch/batch_pool.go

```diff
@@ -363,11 +361,27 @@ func expandBatchPoolContainerRegistry(ref map[string]interface{}) (*batch.Contai
 		return nil, fmt.Errorf("Error: container registry reference should be defined")
 	}
 
-	containerRegistry := batch.ContainerRegistry{
-		RegistryServer: utils.String(ref["registry_server"].(string)),
-		UserName:       utils.String(ref["user_name"].(string)),
-		Password:       utils.String(ref["password"].(string)),
+	containerRegistry := batch.ContainerRegistry{}
+
+	if ref["registry_server"] != nil && ref["registry_server"] != "" {
+		refRegistryServer := ref["registry_server"].(string)
+		containerRegistry.RegistryServer = &refRegistryServer
+	}
```

Comment:
minor: could we change these to be something more like this
```suggestion
	if v := ref["registry_server"]; v != nil && v != "" {
		containerRegistry.RegistryServer = pointer.FromString(v.(string))
	}
```